### PR TITLE
Cache Github releases during docgen

### PR DESCRIPTION
**Pull request by saiskee**
# Description

Please include a summary of the changes.

This bug fixes ... \ This new feature can be used to ...

# Context

Users ran into this bug doing ... \ Users needed this feature to ...

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -125,6 +125,8 @@ replace (
 	// Required for proper serialization of CRDs
 	github.com/renstrom/dedent => github.com/lithammer/dedent v1.0.0
 
+	github.com/solo-io/go-utils => /Users/sai/go/src/github.com/solo-io/go-utils
```
Comment (saiskee) [inline go.mod:128]: Remove this once go-utils PR is merged.

File: docs/cmd/generate_docs.go [hunk 2]
```diff
@@ -50,15 +44,63 @@ func rootApp(ctx context.Context) *cobra.Command {
 	}
 	app.AddCommand(changelogMdFromGithubCmd(opts))
 	app.AddCommand(securityScanMdFromCmd(opts))
+	app.AddCommand(getReleases(opts))
 
-	app.PersistentFlags().StringVar(&opts.HugoDataSoloOpts.version, "version", "", "version of docs and code")
```
Comment (ben-taussig-solo) [inline docs/cmd/generate_docs.go:54]: Good call on removing these -- I was super confused when I tried to pass a version flag into some of the commands defined in this file

File: docs/cmd/generate_docs.go [hunk 3]
```diff
@@ -50,15 +44,63 @@ func rootApp(ctx context.Context) *cobra.Command {
...
+	return app
+}
 
+// Serializes github repository release and prints serialized releases to stdout
+// To be used for caching release data for changelog/security scan docsgen.
+func getReleases(opts *options) *cobra.Command {
+	app := &cobra.Command{
+		Use:   "gen-releases",
+		Short: "cache github releases for gloo edge repository",
+		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (ben-taussig-solo) [inline docs/cmd/generate_docs.go:58]: I think ideally most of this `RunE` function would extracted into its own method to better match the other commands defined in this file (see example below)
https://github.com/solo-io/gloo/blob/d5e1d4624def275227b3075cc2a5169c94dc1086/docs/cmd/generate_docs.go#L104-L115

I think it makes sense to have the `if !useCachedReleases()` clause and `switch target` and `if len(args) != 1`sections at this level, but how would you feel about moving the creation of the github client (lines 63-71) into getRepoReleases?

File: docs/cmd/generate_docs.go [hunk 4]
```diff
@@ -50,15 +44,63 @@ func rootApp(ctx context.Context) *cobra.Command {
...
+				return InvalidInputError(target)
+			}
+			return err
+		},
+	}
 	return app
 }
 
+// Serialized github RepositoryRelease array to be written to file
+func getRepoReleases(ctx context.Context, repo string, client *github.Client) error {
```
Comment (ben-taussig-solo) [inline docs/cmd/generate_docs.go:92]: nit: I think this file will be more readable if `getRepoReleases` is defined after `securityScanMdFromCmd` and `changelogMdFromGithubCmd`, that way all the commands are defined right next to each other

**Review by ben-taussig-solo — APPROVED** [review]
LGTM!

**Pull request merged**